### ISSUE TITLE
window.DlgCode docs

### ISSUE THREAD

**Issue by marc2k3** [TheQwertiest/foo_spider_monkey_panel#51]
The example is all wrong because it's based on my old and wrong docs!! 

It's a `property` but the example makes it look like a `method`...

```js
window.DlgCode(DLGC_WANTALLKEYS);
```

It should be

```js
window.DlgCode = DLGC_WANTALLKEYS;
```

As you may or may not recall, it's used in conjunction with `Grab focus` for the keyboard related callbacks. I remember you had an issue with your playlist not even using it and it worked fine in CUI and not DUI. This is because Columns UI defaults to returning `DLGC_WANTALLKEYS` using [windows messaging](https://github.com/reupen/columns_ui/blob/fac85cbc3e8019a1d9b5f523d2143eeb53590df5/foo_ui_columns/mw_wnd_proc.cpp#L479) whereas DUI returns zero and that's you have to specify it manually.

I've actually nuked the whole thing in my latest release and made DUI behave the same as CUI without having to specify it.... of course this breaks scripts!!

https://hydrogenaud.io/index.php/topic,112914.msg967255.html#msg967255

Obviously it's mentioned on the releases page, in the changelog and it's a BETA but no one ever reads that crap!!

**Comment by TheQwertiest** [comment]
> The example is all wrong because it's based on my old and wrong docs!!

Thanks!

> I've actually nuked the whole thing in my latest release and made DUI behave the same as CUI without having to specify it

Yeah, I was planning to adapt your changes in v2.0.0 release (which will bring other API incompatibilities as well), since I want to maintain API compatibility in v1.x.x (semantic versioning and stuff).

> Obviously it's mentioned on the releases page, in the changelog and it's a BETA but no one ever reads that crap!!

I know, right? It becomes really frustrating sometimes -_-
Some PM's I've received made me really disappointed in humanity... Makes me that much more appreciative of the folks who can think for themselves though :D

**Comment by marc2k3** [comment]
Of course I could have the thing a no op so as not to break scripts but you know what I'm like... I'm a terrible amateur who enjoys breaking things more than I do adding new features!!

**Comment by TheQwertiest** [comment]
Thanks! That reminds me: I've actually introduced a memory leak in JScript via `setTimeout`/`setInterval` while overhauling it. Your repository was archived at the time, so I forgot about it... Since you are resuming the development of JScript, I will fix it in a pull request in the near future. Sorry for that -_-""

**Comment by Wil-B** [comment]
For forward compatibility is it safe / best to add something like the following now?

if ('DlgCode' in window) { window.DlgCode = 4; }

**Comment by TheQwertiest** [comment]
@Wil-B : If you are talking about SMP, then I don't think there is a need for this workaround - v2.0.0 *will* bring several API breaking changes, but it's still too far in the future to worry about it. The only thing you should keep an eye out is API marked as **deprecated** =)

**Comment by TheQwertiest** [comment]
> I'd consider removing Grab focus as a panel option and turning it off by default

This will be incorporated (one way or another) in the future Configuration dialog overhaul (which is currently planned to be implemented in v1.3).
